File: jenjin-core/src/main/java/com/binarymonks/jj/physics/collisions/SoundCollision.java
package com.binarymonks.jj.physics.collisions;

import com.badlogic.gdx.physics.box2d.Contact;
import com.badlogic.gdx.physics.box2d.Fixture;
import com.binarymonks.jj.audio.SoundMode;
import com.binarymonks.jj.physics.CollisionFunction;
import com.binarymonks.jj.things.Thing;

public class SoundCollision extends CollisionFunction {

    String soundID;
    SoundMode mode = SoundMode.NORMAL;

    public SoundCollision(String soundID, SoundMode mode) {
        this.soundID = soundID;
        this.mode = mode;
    }

    public SoundCollision(String soundID) {
        this.soundID = soundID;
    }

    @Override
    public void collision(Thing me, Fixture myFixture, Thing other, Fixture otherFixture, Contact contact) {
        me.sounds.triggerSound(soundID, mode);
    }

    @Override
    public CollisionFunction clone() {
        return new SoundCollision(soundID, mode);
    }
}

File: jenjin-core/src/main/kotlin/com/binarymonks/jj/core/physics/collisions/SoundCollision.kt
package com.binarymonks.jj.core.physics.collisions

import com.badlogic.gdx.physics.box2d.Contact
import com.badlogic.gdx.physics.box2d.Fixture
import com.binarymonks.jj.core.audio.SoundMode
import com.binarymonks.jj.core.physics.CollisionHandler
import com.binarymonks.jj.core.scenes.Scene

/**
 * Triggers a sound on collision
 */
class SoundCollision(
        var soundName: String? = null,
        var mode: SoundMode = SoundMode.NORMAL
) : CollisionHandler() {

    override fun collision(me: Scene, myFixture: Fixture, other: Scene, otherFixture: Fixture, contact: Contact) {
        me.soundEffects.triggerSound(checkNotNull(soundName), mode)
    }

    override fun clone(): CollisionHandler {
        val copy = SoundCollision()
        copy.soundName = soundName
        copy.mode = mode
        copy.copyProperties(this)
        return copy
    }
}